src/containers/Network/NodeSetup/ClientConfigForm.jsx

```diff
@@ -0,0 +1,246 @@
+import React, { Component } from 'react'
+import { Input, FileChooser, Select } from 'ethereum-react-components'
+import styled, { css } from 'styled-components'
+import { Mist } from '../../../API'
+
+const { geth } = Mist
+
+export default class ClientConfigForm extends Component {
+  state = {
+    config: {},
+    // status: {},
+    options: {
+      networks: ['main', 'ropsten', 'rinkeby'],
+      ipcModes: ['IPC', 'WebSockets'],
+      syncModes: ['light', 'fast', 'full']
+    }
+  }
+
+  componentDidMount() {
+    this.getStatus()
+    this.getConfig()
+  }
+
+  async setConfig(config) {
+    await geth.setConfig({ config })
+    this.setState({ config })
+  }
+
+  // async getStatus() {
+  //   const status = await geth.getStatus()
+  //   this.setState({ status })
+  // }
+
+  async getConfig() {
+    const config = await geth.getConfig()
+    this.setState({ config })
+  }
+
+  handleChangeDataDir = event => {
+    // event.target.value returns `fakepath` in electron due to security reasons.
+    // Would need to use electron's remote dialog.showOpenDialog()
+    const { config } = this.state
+    const newConfig = { ...config, dataDir: event.target.value }
+    this.setState({ config: newConfig })
+    geth.setConfig(newConfig)
+  }
+
+  handleChangeSyncMode = selectedOption => {
+    const { config } = this.state
+    const syncMode = selectedOption.value
+    const newConfig = { ...config, syncMode }
+    this.setState({ config: newConfig })
+    geth.setConfig(newConfig)
+  }
+
+  handleChangeIpc = selectedOption => {
+    const { config } = this.state
+    const ipc = selectedOption.value
+    const newConfig = { ...config, ipc }
+    this.setState({ config: newConfig })
+    geth.setConfig(newConfig)
+  }
+
+  handleChangeNetwork = selectedOption => {
+    const { config } = this.state
+    const network = selectedOption.value
+    const newConfig = { ...config, network }
+    this.setState({ config: newConfig })
+    geth.setConfig(newConfig)
+  }
+
+  handleChangeHost = event => {
+    const { config } = this.state
+    const newConfig = { ...config, host: event.target.value }
+    this.setState({ config: newConfig })
+    geth.setConfig(newConfig)
+  }
+
+  handleChangePort = event => {
+    const { config } = this.state
+    const newConfig = { ...config, port: Number(event.target.value) }
+    this.setState({ config: newConfig })
+    geth.setConfig(newConfig)
```

Comment:
promoting one Source of Truth™, `geth.setConfig` could be called only when start/stop btn clicked.

Reply:
ty